File: core-java-practice/leet-code-codebase/reverseinteger.java
public class reverseinteger {
    public static void main(String[] args) {
        int x = 123;
        int reversed = 0;
        while (x != 0) {
            int digit = x % 10;
            // Check for overflow
            if (reversed > (Integer.MAX_VALUE - digit) / 10) {
                System.out.println(0);
                return;
            }
            reversed = reversed * 10 + digit;
            x /= 10;
        }
        System.out.println(reversed);
    }
}

File: core-csharp-practice/leet-code-practice/ReverseInteger.cs
using System;

class ReverseInteger
{
    static void Main()
    {
        int x = 123;   // fixed number
        int reversed = 0;

        while (x != 0)
        {
            int digit = x % 10;

            // Overflow check
            if (reversed > (int.MaxValue - digit) / 10)
            {
                Console.WriteLine(0);
                return;
            }

            reversed = reversed * 10 + digit;
            x /= 10;
        }

        Console.WriteLine(reversed);
    }
}
